core/src/main/java/org/elasticsearch/tribe/TribeService.java

```diff
@@ -463,6 +487,44 @@ private ClusterState applyUpdate(ClusterState currentState, ClusterChangedEvent
                 }
             }
 
+            Set<String> changedCustomMetaDataTypeSet = tasks.stream()
+                    .map(ClusterChangedEvent::changedCustomMetaDataSet)
+                    .flatMap(Collection::stream)
+                    .collect(Collectors.toSet());
+            final List<Node> tribeNodes = TribeService.this.nodes;
+            Map<String, MetaData.Custom> mergedCustomMetaDataMap = mergeChangedCustomMetaData(changedCustomMetaDataTypeSet,
+                    customMetaDataType -> {
+                        List<MetaData.Custom> tribeCustomMetaDataList = new ArrayList<>(tribeNodes.size());
+                        for (Node tribeNode : tribeNodes) {
+                            String currentTribeName = TRIBE_NAME_SETTING.get(tribeNode.settings());
+                            final MetaData.Custom custom;
+                            if (currentTribeName.equals(tribeName)) {
+                                // current tribe node
+                                custom = tribeState.metaData().custom(customMetaDataType);
```

Comment:
Apologies for the misleading comment (updated it now). This special cases getting the custom for the tribeClient that triggered the current cluster state update. Because we have the latest cluster state from the `clusterChangedEvent` for this tribeClient, we use that instead of looking up the state from the tribeClient's cluster service. 
I think this is the right thing to do here, though I have tested it out without this special case and it still works. I am worried if the tribeClient's state from the cluster service (which initiated the cluster state update) is outdated w.r.t its latest changes. WDYT @bleskes ?